### PR TITLE
[GH-25494] add request context to public methods in bot go

### PR DESCRIPTION
**Pull request by ezekielchow**
#### Summary
- Added request context as a parameter to public methods in bots.go
- Added request parameter to functions that call bots.go methods which requires request context
- Rename c to rctx to signify request context

#### Ticket Link
https://github.com/mattermost/mattermost/issues/25494

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
Nicely done, thanks @ezekielchow :+1:

File: server/channels/app/authorization.go [hunk 1]
```diff
@@ -339,8 +339,9 @@ func (a *App) RolesGrantPermission(roleNames []string, permissionId string) bool
 // SessionHasPermissionToManageBot returns nil if the session has access to manage the given bot.
 // This function deviates from other authorization checks in returning an error instead of just
 // a boolean, allowing the permission failure to be exposed with more granularity.
-func (a *App) SessionHasPermissionToManageBot(session model.Session, botUserId string) *model.AppError {
-	existingBot, err := a.GetBot(botUserId, true)
+func (a *App) SessionHasPermissionToManageBot(rctx request.CTX, session model.Session, botUserId string) *model.AppError {
+	// TODO: Ezekiel
```
[the remark this answers is not in the document]
Comment (ezekielchow) [reply server/channels/app/authorization.go]: Removed

**Review by fmartingr — APPROVED** [review]

**Comment by hanzei** [comment]
Thanks for the contribution @ezekielchow!

**Comment by hanzei** [comment]
@amyblais There is no need to have this PR in v9.7. v9.8 is fine.

**Pull request merged**